optimade/server/config.py

```diff
@@ -1,260 +1,68 @@
-            raise NotImplementedError(
-                f"load function for config format {ftype} is not implemented"
-            )
-        else:
-            f()
-
-    @property
-    def debug(self):
-        return self._debug
-
-    @debug.setter
-    def debug(self, value: bool):
-        self._debug = value
-
-
-class ServerConfig(Config):
+class ServerConfig(BaseSettings):
     """ This class stores server config parameters in a way that
     can be easily extended for new config file types.
 
     """
 
-    @staticmethod
-    def _DEFAULTS(field: str) -> Any:
-        res = {
-            "use_real_mongo": False,
-            "mongo_database": "optimade",
-            "mongo_uri": "localhost:27017",
-            "links_collection": "links",
-            "references_collection": "references",
-            "structures_collection": "structures",
-            "page_limit": 20,
-            "page_limit_max": 500,
-            "default_db": "test_server",
-            "base_url": None,
-            "implementation": {
-                "name": "Example implementation",
-                "version": __version__,
-                "source_url": "https://github.com/Materials-Consortia/optimade-python-tools",
-                "maintainer": None,
-            },
-            "provider": {
-                "prefix": "_exmpl_",
-                "name": "Example provider",
-                "description": "Provider used for examples, not to be assigned to a real database",
-                "homepage": "https://example.com",
-                "index_base_url": None,
-            },
+    debug: bool = False
+    use_real_mongo: bool = False
+    mongo_database: str = "optimade"
+    mongo_uri: str = "localhost:27017"
+    links_collection: str = "links"
+    references_collection: str = "references"
+    structures_collection: str = "structures"
+    page_limit: int = 20
+    page_limit_max: int = 500
+    default_db: str = "test_server"
+    base_url: Optional[str] = None
+    implementation: Implementation = Implementation(
+        name="Example implementation",
+        version=__version__,
+        source_url="https://github.com/Materials-Consortia/optimade-python-tools",
+        maintainer=None,
+    )
+    provider: Provider = Provider(
+        prefix="_exmpl_",
+        name="Example provider",
+        description="Provider used for examples, not to be assigned to a real database",
+        homepage="https://example.com",
+        index_base_url="http://localhost:5001",
+    )
+    provider_fields: Dict[Literal["links", "references", "structures"], List[str]] = {
+        "structures": ["band_gap", "_mp_chemsys"]
+    }
+    aliases: Dict[Literal["links", "references", "structures"], Dict[str, str]] = {
+        "structures": {
+            "id": "task_id",
+            "chemical_formula_descriptive": "pretty_formula",
+            "chemical_formula_reduced": "pretty_formula",
+            "chemical_formula_anonymous": "formula_anonymous",
```

Comment:
Could we all chip in here to set these up as pydantic `Field`s with a description? It's unclear to me at the moment if we're still using e.g. the `base_url` option!

Reply:
`base_url` is necessary for the openapi spec, but it might just be the same as something else. I'd rather not co-mingle eliminating variables with this PR.

Reply:
`base_url` is very useful for us, since we have several different implementations / databases that we want to serve, each with a different base URL that needs to be incorporated into the pageing links and more.